src/main/kotlin/no/skatteetaten/aurora/boober/service/GitService.kt

```diff
@@ -67,21 +69,42 @@ open class GitService(
 
     val cp = UsernamePasswordCredentialsProvider(username, password)
 
-    fun checkoutRepository(repositoryName: String): Git {
-        val repoPath = File(File("$checkoutPath/$repositoryName").absoluteFile.absolutePath)
-        if (repoPath.exists()) {
+    @JvmOverloads
+    fun checkoutRepository(repositoryName: String, checkoutFolder: String = repositoryName, deleteUnpushedCommits: Boolean = true): Git {
+        val repoPath = File(File("$checkoutPath/$checkoutFolder").absoluteFile.absolutePath)
+        val git = if (repoPath.exists()) {
```

Comment:
Jeg foretrekker å unngå null hvis jeg kan. Et forslag til å unngå det her er å ha en guard på `!repoPath.exists()` for så å trekke ut og returne direkte fra try/catch.